dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/SamplesToVariantSourceProcessor.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.jobs.steps.processors;
+
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.commons.core.models.Aggregation;
+import uk.ac.ebi.eva.commons.core.models.IVariantSource;
+import uk.ac.ebi.eva.commons.core.models.StudyType;
+import uk.ac.ebi.eva.commons.core.models.VariantSource;
+import uk.ac.ebi.eva.dbsnpimporter.models.Sample;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Packs a set of {@link Sample} under a {@link uk.ac.ebi.eva.commons.core.models.IVariantSource} object, adding some
+ * additional batch information.
+ */
+public class SamplesToVariantSourceProcessor implements ItemProcessor<List<Sample>, IVariantSource> {
+
+    public static final String DBSNP_BUILD_KEY = "dbsnp-build";
+
+    public static final String DBSNP_BATCH_KEY = "dbsnp-batch-id";
+
+    private final String dbsnpBuild;
+
+    private final String batchId;
+
+    public SamplesToVariantSourceProcessor(int dbsnpBuild, int batchId) {
+        this.dbsnpBuild = String.valueOf(dbsnpBuild);
+        this.batchId = String.valueOf(batchId);
+    }
+
+    @Override
+    public IVariantSource process(List<Sample> samples) throws Exception {
+        if (areSamplesFromMultipleBatches(samples)) {
+            throw new IllegalArgumentException("Samples must belong to a single batch");
+        }
+
+        if (isSampleDuplicated(samples)) {
+            throw new IllegalArgumentException("The batch contains duplicate samples");
+        }
+
+        // Study ID, file ID, study name, file name
+        String batchName = samples.get(0).getBatch();
+        String studyId = batchName;
```

Comment:
loc_batch_id_upp is not forced to be unique in the database. Are we sure we are not going to have any clash between different batches with the same name?

Reply:
@sundarvenkata-EBI ran checks on several databases and there weren't clashes in any of them.